src/main/kotlin/io/libp2p/protocol/ProtocolMessageHandler.kt

```diff
@@ -0,0 +1,15 @@
+package io.libp2p.protocol
+
+import io.libp2p.core.Stream
+
+interface ProtocolMessageHandler<TMessage> {
+    fun onActivated(stream: Stream) = Unit
+    fun onMessage(stream: Stream, msg: TMessage) = Unit
+    fun onClosed(stream: Stream) = Unit
+    fun onException(cause: Throwable?) = Unit
+
+    fun fireMessage(stream: Stream, msg: Any) {
+        @Suppress("UNCHECKED_CAST")
+        onMessage(stream, msg as TMessage)
+    }
```

Comment:
Is this method needed in public interface?

Reply:
It does, because it needs to be accessible from ProtocolMessageHandlerAdapter. Ideally it would be internal, but unfortunately that's not available on interface methods.